lumicks/pylake/conftest.py

```diff
@@ -184,27 +218,68 @@ def get_reference_data(reference_data, test_name=None, file_name=None):
         calling_path = request.path.parent
 
         reference_data_path = calling_path / "reference_data" / calling_function
-        reference_file_path = reference_data_path / f"{ref_data_filename}.npz"
+        reference_file_path = reference_data_path / f"{ref_data_filename}.{extension}"
 
         if reference_file_path.exists() and not request.config.getoption("--update_reference_data"):
-            with np.load(reference_file_path, allow_pickle=True) as npz_file:
-                return npz_file["arr_0"][()]
+            return read_data(reference_file_path)
         else:
             if request.config.getoption("--strict_reference_data"):
                 raise RuntimeError(
                     f"Test data for {calling_function}/{ref_data_filename} missing! Did you forget "
-                    f"to check it in?"
+                    f"to check the file {reference_file_path} in?"
                 )
 
             reference_data_path.mkdir(parents=True, exist_ok=True)
+            write_data(reference_file_path, reference_data)
 
-            try:
-                reference_data = np.asarray(reference_data)
-            except ValueError:
-                reference_data = np.asarray(reference_data, dtype=object)
-
-            np.savez(reference_file_path, reference_data)
             print(f"\nWritten reference data {ref_data_filename} to {reference_file_path}.")
             return reference_data
 
     return get_reference_data
+
+
+@pytest.fixture
+def compare_to_reference_dict(request, reference_data):
```

Comment:
Is the `request` fixture needed here?

Reply:
Good catch! It was required in a previous implementation, but isn't anymore.